app/src/gui/components/notebook/record_table.tsx

```diff
@@ -448,29 +439,45 @@ function RecordsTable(props: RecordsTableProps) {
   );
 }
 
+interface RecordMetaDataExtended extends RecordMetadata {
+  record: Record;
+}
+
 export function RecordsBrowseTable(props: RecordsBrowseTableProps) {
   const [query, setQuery] = React.useState('');
-  const [pouchData, setPouchData] = React.useState(
-    undefined as RecordMetadata[] | undefined
-  );
+  const [pouchData, setPouchData] = React.useState<
+    RecordMetaDataExtended[] | undefined
+  >(undefined);
 
   useEffect(() => {
     const getData = async () => {
       try {
-        if (query.length === 0) {
-          const ma = await getMetadataForAllRecords(
-            props.project_id,
-            props.filter_deleted
-          );
-          setPouchData(ma);
-        } else {
-          const ra = await getRecordsWithRegex(
-            props.project_id,
-            query,
-            props.filter_deleted
+        const records = (
+          query.length === 0
+            ? await getMetadataForAllRecords(
+                props.project_id,
+                props.filter_deleted
+              )
+            : await getRecordsWithRegex(
+                props.project_id,
+                query,
+                props.filter_deleted
+              )
+        ) as RecordMetaDataExtended[];
+
+        for (const record of records) {
+          const data = await getFullRecordData(
+            record.project_id,
+            record.record_id,
+            record.revision_id
           );
```

Comment:
Performance concerns here possibly - but we'll see.

Reply:
For the PR to display summary fields in the record table we'll be grabbing the whole record data so this would not be necessary. A few interacting changes going on here. Can this be deferred?